### PR TITLE
Two more fixes for notification memory leaks

### PR DESCRIPTION
**Pull request by swick**

### CODE REVIEW

File: libportal/notification.c [hunk 1]
```diff
@@ -580,7 +585,7 @@ parse_notification_cb (GObject      *source_object,
   XdpPortal *portal = g_task_get_source_object (task);
   g_autoptr(GError) error = NULL;
   g_autoptr(GUnixFDList) fd_list = NULL;
-  g_autoptr(GVariant) parameters = NULL;
+  GVariant *parameters = NULL;
```
Comment (GeorgesStavracas) [inline libportal/notification.c:588]: Wouldn't it be more semantic to keep the `g_autoptr` here, and make `parse_notification_finish()` return `g_steal_pointer (&data->data);`?

**Comment by jsparber** [comment]
Thansk for looking into this. Lgtm

**Review by GeorgesStavracas — APPROVED** [review]
One suggestion but overall it LGTM

File: libportal/notification.c (hunk 1 above)
Comment (swick) [reply libportal/notification.c]: Good point.

**Pull request merged**